crates/wasmparser/src/validator/core.rs

```diff
@@ -876,6 +876,26 @@ impl Module {
         offset: usize,
     ) -> Result<()> {
         self.check_limits(ty.initial, ty.maximum, offset)?;
+        let page_size = if let Some(page_size_log2) = ty.page_size_log2 {
+            if !features.custom_page_sizes {
+                return Err(BinaryReaderError::new(
+                    "the custom page sizes proposal must be enabled to \
+                     customize a memory's page size",
+                    offset,
+                ));
+            }
+            let page_size = 1_u64 << page_size_log2;
+            debug_assert!(page_size.is_power_of_two());
+            if page_size > DEFAULT_WASM_PAGE_SIZE {
+                return Err(BinaryReaderError::new(
+                    "invalid custom page size: {page_size}",
+                    offset,
+                ));
+            }
+            page_size
```

Comment:
I think this will want to validate that `page_size_log2` is <64 first or otherwise I think the over-wide shift will panic in debug mode

Reply:
(and might as well bundle in the check that it's less than 16 to begin with)